### PR TITLE
[SPARK-44539][BUILD] Upgrade RoaringBitmap to 1.0.0

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
- The pr aims to upgrade RoaringBitmap from 0.9.45 to 1.0.0.
- From version 1.0.0, the `ArraysShim` class has been moved from `shims-x.x.x.jar` jar to `RoaringBitmap-x.x.x.jar` jar, so we no longer need to rely on it.

### Why are the changes needed?
- The newest brings some improvments, eg:
Add zero-garbage deserialiser for ByteBuffer to RoaringBitmap by @shikharid in https://github.com/RoaringBitmap/RoaringBitmap/pull/650
More specialized method for value decrementation by @xtonik in https://github.com/RoaringBitmap/RoaringBitmap/pull/640
Duplicated small array sort routine by @xtonik in https://github.com/RoaringBitmap/RoaringBitmap/pull/638
Avoid intermediate byte array creation by @xtonik in https://github.com/RoaringBitmap/RoaringBitmap/pull/635
Useless back and forth BD bytes conversion by @xtonik in https://github.com/RoaringBitmap/RoaringBitmap/pull/636

- The full release notes:
https://github.com/RoaringBitmap/RoaringBitmap/releases/tag/1.0.0
https://github.com/RoaringBitmap/RoaringBitmap/releases/tag/0.9.49
https://github.com/RoaringBitmap/RoaringBitmap/releases/tag/0.9.48
https://github.com/RoaringBitmap/RoaringBitmap/releases/tag/0.9.47
https://github.com/RoaringBitmap/RoaringBitmap/releases/tag/0.9.46

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### CODE REVIEW

**Comment by panbingkun** [comment]
Let me add the results of org.apache.spark.MapStatusesConvertBenchmark.

File: dev/deps/spark-deps-hadoop-3-hive-2.3 [hunk 1]
```diff
@@ -1,7 +1,7 @@
 HikariCP/2.5.1//HikariCP-2.5.1.jar
 JLargeArrays/1.5//JLargeArrays-1.5.jar
 JTransforms/3.1//JTransforms-3.1.jar
-RoaringBitmap/0.9.45//RoaringBitmap-0.9.45.jar
+RoaringBitmap/0.9.46//RoaringBitmap-0.9.46.jar
```
Comment (HyukjinKwon) [inline dev/deps/spark-deps-hadoop-3-hive-2.3:4]: I think we should probably have a buferred update within individual release thread. Upgrading is good and a legitimate improvement but could be done in batch within each release window.

Comment (yaooqinn) [reply dev/deps/spark-deps-hadoop-3-hive-2.3:4]: +1 to @HyukjinKwon's

Comment (panbingkun) [reply dev/deps/spark-deps-hadoop-3-hive-2.3:4]: Maybe once a month? Or two months?

**Comment by panbingkun** [comment]
From version 1.0.0, the `ArraysShim` class has been moved from `shims-x.x.x.jar` jar to `RoaringBitmap-x.x.x.jar` jar, so we no longer need to rely on it.

image

image

**Comment by panbingkun** [comment]
@HyukjinKwon
After the official release of version 1.0.0, I think we should upgrade this major version, which includes some changes that rely on jars.

cc @dongjoon-hyun @LuciferYang

**Review by LuciferYang — APPROVED** [review]
LGTM from my side

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**